docs/api/app.md

```diff
@@ -309,6 +309,39 @@ app.on('certificate-error', (event, webContents, url, error, certificate, callba
 })
 ```
 
+### Event: 'client-certificate-request-password' _Linux_
+
+Returns:
+
+* `event` Event
+* `hostName` string - the hostname of the site requiring a client certificate
+* `tokenName` string - the token (or slot) name of the cryptographic device
+* `retry` boolean - whether there have been failed attempt at prompting the password
+
+Emitted when a password is needed to unlock a client certificate for
+`hostName`. You need to call `event.preventDefault()`, prompt the user for the
```

Comment:
It would be better not to require `event.preventDefault()` here I think; it's a bit confusing to keep track of when to call or not call `preventDefault` and `callback`. In fact, this API doesn't fit super well with the "event" style; it's really more of a "handler", in the vein of `setWindowOpenHandler`. In general, I think a "handler"-style API makes more sense when:
1. There isn't really any reason to have multiple listeners attached to the event, and
2. A response is required (not optional) from the JS side.

Again, historically not all of Electron's APIs abide by these rules, but we can do better for new APIs :)